typed-racket-lib/typed-racket/typecheck/tc-structs.rkt

```diff
@@ -433,41 +470,43 @@
                      maybe-parsed-proc-ty]
                     [else (expected-but-got top-func maybe-parsed-proc-ty)
                           #f]))))
-         
+
          (define parent-mutable
            ;; Only valid as long as typed structs must be
            ;; either fully mutable or fully immutable
            (or (not parent)
                (andmap fld-mutable? (get-flds concrete-parent))))
-         
+
          (define desc (struct-desc
                        (map fld-t (get-flds concrete-parent))
                        types
                        tvars
                        mutable
                        parent-mutable
-                       maybe-proc-ty))
+                       maybe-proc-ty
+                       property-tys))
          (define sty (mk/inner-struct-type names desc concrete-parent))
-         
+
          (parsed-struct sty names desc (struct-info-property nm/par) type-only)]))
 
 ;; register a struct type
 ;; convenience function for built-in structs
 ;; tc/builtin-struct : identifier Maybe[identifier] Listof[identifier]
-;;                     Listof[Type] Maybe[identifier] Listof[Type]
+;;                     Listof[Type] Maybe[identifier] Listof[Type] ListOf[Pairof[identifier, Type]]
```

Comment:
Only need a list of the property names, but they are expressions (not just ids).